### PR TITLE
Add unit tests for pack util

### PR DESCRIPTION
**Pull request by CongMa13**

### CODE REVIEW

**Comment by CongMa13** [comment]
[ RUN      ] KernelTests/PackUtilTest.RunKernel/62
w64, bf16, 64[       OK ] KernelTests/PackUtilTest.RunKernel/62 (0 ms)
[ RUN      ] KernelTests/PackUtilTest.RunKernel/63
w64, f16, 64[       OK ] KernelTests/PackUtilTest.RunKernel/63 (0 ms)
[ RUN      ] KernelTests/PackUtilTest.RunKernel/64
w64, h16, 64[       OK ] KernelTests/PackUtilTest.RunKernel/64 (0 ms)
[ RUN      ] KernelTests/PackUtilTest.RunKernel/65
w64, f32, 64[       OK ] KernelTests/PackUtilTest.RunKernel/65 (0 ms)
[ RUN      ] KernelTests/PackUtilTest.RunKernel/66
w64, xf32, 64[       OK ] KernelTests/PackUtilTest.RunKernel/66 (0 ms)
[ RUN      ] KernelTests/PackUtilTest.RunKernel/67
w64, i8, 64[       OK ] KernelTests/PackUtilTest.RunKernel/67 (0 ms)
[ RUN      ] KernelTests/PackUtilTest.RunKernel/68
w64, i32, 64[       OK ] KernelTests/PackUtilTest.RunKernel/68 (0 ms)
[ RUN      ] KernelTests/PackUtilTest.RunKernel/69
w64, f64, 64[       OK ] KernelTests/PackUtilTest.RunKernel/69 (0 ms)
[----------] 70 tests from KernelTests/PackUtilTest (449 ms total)

[----------] Global test environment tear-down
[==========] 70 tests from 1 test suite ran. (449 ms total)
[  PASSED  ] 56 tests.

File: test/unit/pack_util_test/device/pack_util.hpp [hunk 1]
```diff
@@ -66,6 +66,314 @@ namespace rocwmma
...
+            using UnpackedT = typename PackUtil::Traits::UnpackedT;
+
+            auto paddedData = PackUtil::pad(res);
+            err |= !std::is_same_v<decltype(paddedData), VecT<PackedT, VecSize>>;
+            for(uint32_t i = 0; i < VecSize; i++)
+            {
+                auto value = get(paddedData, i);
+
+                // unpadded data is at idx 0 in the padded data by default
+                err |= *reinterpret_cast<DataT*>(&value) != static_cast<DataT>((float)i);
```
Comment (cgmillette) [inline test/unit/pack_util_test/device/pack_util.hpp:99]: Is the float cast to get around the integer ctor of bfloat16?

Comment (CongMa13) [reply test/unit/pack_util_test/device/pack_util.hpp:99]: yeah, building was failed since uint32_t cannot be cast to bf16.

**Review by cgmillette — APPROVED** [review]
LGTM!

**Review by dlangbe — APPROVED** [review]

**Pull request merged**